src/RequestHeaders.cs

```diff
@@ -20,7 +21,9 @@ public class RequestHeaders : IDictionary<string,IEnumerable<string>> {
             throw new ArgumentNullException(nameof(headerValues));
         if(!headerValues.Any())
             return;
-        if(_headers.TryGetValue(headerName, out var values))
+        if(_singleValueHeaders.Contains(headerName) && _headers.ContainsKey(headerName) && _headers[headerName].Any())
+            return;
```

Comment:
I believe we want a replacement of the original header. 
See https://github.com/microsoftgraph/msgraph-sdk-dotnet/issues/1951#issuecomment-1588649698